Sources/MongoSwift/APM.swift

```diff
@@ -164,7 +164,7 @@ public struct ServerDescriptionChangedEvent: MongoEvent, InitializableFromOpaque
         self.connectionId = ConnectionId(mongoc_apm_server_changed_get_host(event))
         var oid = bson_oid_t()
         mongoc_apm_server_changed_get_topology_id(event, &oid)
-        self.topologyId = ObjectId(fromPointer: &oid)
+        self.topologyId = ObjectId(copying: &oid)
```

Comment:
Consider adhering to the lessons learned with the Swift 5 DocumentIterator change, and switch to using `with*Pointer` methods here:
```
var oid = bson_oid_t()
withUnsafeMutablePointer(to: oid) { oidPtr in
  mongoc_apm_server_changed_get_topology_id(event, oidPtr)
  self.topologyId = ObjectId(copying: oidPtr)
...
```

I think this makes the access more explicit, as well as the expected lifetime

Reply:
I see a few recurrences of this pattern below, so I'd say if we change it here let's do it everywhere

Reply:
I've updated all the `get_topology_id` calls to use `withUnsafeMutablePointer` here and below.

I've also just gone ahead and removed the pointer initializer and replaced it with one that just takes in a straight `bson_oid_t`. We can just pass the  `bson_oid_t` in by value.

Reply:
Nice!